### DISCUSSION TITLE
SSR+Typescript setup

### DISCUSSION THREAD

**Discussion by Asxer** [vuejs/pinia#1742] [Help and Questions]
Hello! I'm in the process of exploring Nuxt 3 and Pinia and I'm very confused right now. Could you please help me? 

1. I see in the documentation words 
 > Proper TypeScript support
But I don't see any Typescript support at all. I always get errors from IDE that some actions don't exist and there is no method $patch. I can not find defined generic types for stores to use like this for example:

```
const store: Store<AuthState> = useAuthStore();

store.$patch({ some: 'value' });

store.login();
```
Where can I find types set for the Pinia?

2. I tried to use `@pinia/nuxt` but there are constant errors about `Can not find module pinia` so I had to use such a workaround in nuxt.config.js
```
alias: {
    pinia: '/node_modules/@pinia/nuxt/node_modules/pinia/dist/pinia.mjs',
  }
```
I don't really like it and due to this problem, there is no autocomplete, proper work with types, etc.

3. I tried to use Pinia in SSR but I don't get at all how to use [this part of the code from the documentation](https://pinia.vuejs.org/ssr/#using-the-store-outside-of-setup)
```
const pinia = createPinia()
const app = createApp(App)
```

What is App? Where should it be imported from? From where should I import the `createPinia` function? Due to point 2, there are no autocompletes for me.

**Comment by thiswallz** [comment]
hi @Asxer !, you should use `@pinia/nuxt` which works for nuxt 2 and 3,  this lib is calling `createPinia()` behind the scenes for you! 

check this setup: https://dev.to/manojap/setup-pinia-in-nuxt-3-37ia
